src/main/java/com/saucelabs/saucerest/SauceREST.java

```
@@ -301,16 +325,250 @@ public void addTags(String jobId, List<String> tags) {
+     * specified by the <code>location</code> field.
+     *
+     * @param jobId      the Sauce job ID, typically equal to the Selenium/WebDriver sessionId
+     * @param location   represents the base directory where the device log should be downloaded to
+     * @param filename   represents the filename to store the content
+     * @param isEmulator to determine what device log to download
+     * @return True if the device log was downloaded successfully; Otherwise false
+     */
+    public boolean downloadDeviceLog(String jobId, String location, String filename, boolean isEmulator) {
+        URL restEndpoint;
+        if (isEmulator) {
+            restEndpoint = this.buildURL(username + "/jobs/" + jobId + "/assets/logcat.log");
+        } else {
+            // isSimulator
+            restEndpoint = this.buildURL(username + "/jobs/" + jobId + "/assets/ios-syslog.log");
+        }
+
+        return saveFile(jobId, location, filename, restEndpoint);
+    }
+
+    /**
+     * Downloads the screenshots for a Sauce job to the filesystem.  The file will be stored in a directory
+     * specified by the <code>location</code> field.
+     * <p>
+     *
+     * @param jobId      the Sauce job ID, typically equal to the Selenium/WebDriver sessionId
+     * @param location   represents the base directory where the device log should be downloaded to
+     * @param isEmulator to determine what device log to download
+     * @throws FileNotFoundException                                if the log is missing or doesn't exist
+     * @throws com.saucelabs.saucerest.SauceException.NotAuthorized if credentials are wrong or missing
+     * @throws IOException                                          if something else goes wrong during asset retrieval
+     */
+    public void downloadDeviceLogOrThrow(String jobId, String location, boolean isEmulator) throws SauceException.NotAuthorized, IOException {
```

Do you know why we have these `OrThrow` methods? They are giving implementation details about the API wrapper. I mean, we should always throw exceptions that the client should handle, and not offer these methods that make them think if they should handle that or not.

---

I don't know why we have `OrThrow`and the others return a `boolean`. In this case I just went with the rest of the code. If I understand you correctly, I agree with you that the method name shouldn't include `OrThrow` and instead always throw in case of an error and otherwise return `true`. I can make that change in all methods in another PR.

---

Some context on orThrow methods:

https://github.com/saucelabs/saucerest-java/pull/90
https://github.com/saucelabs/saucerest-java/issues/107

If I remember correctly before download could happen and the exception would be swallowed without a warning. 
Dylan changed this so a set of methods would throw the exception if needed and other would return a boolean. 

Did I get that right @DylanLacey?

---

I don't think that looks good, mentioning that a method throws in its name is not ideal. We should deprecate these methods and always throw those exceptions, the client should manage them.
But doing that is not in the scope of this PR, so I am ok with it (for now)